Emby.Dlna/PlayTo/Device.cs

```diff
@@ -1127,23 +1134,29 @@ private void OnMediaChanged(uBaseObject old, uBaseObject newMedia)
         #region IDisposable
 
         bool _disposed;
+
         public void Dispose()
         {
-            if (!_disposed)
-            {
-                _disposed = true;
-
-                DisposeTimer();
-            }
+            Dispose(true);
+            GC.SuppressFinalize(this);
         }
 
-        private void DisposeTimer()
+        protected virtual void Dispose(bool disposing)
         {
-            if (_timer != null)
+            if (_disposed)
             {
-                _timer.Dispose();
-                _timer = null;
+                return;
             }
+
+            if (disposing)
+            {
+                _timer?.Dispose();
+            }
+
+            _timer = null;
```

Comment:
I'm no C# expert, but what happens if this is called with `disposing = false`, and you set `_timer` to `null`, and then another call comes with `disposing = true`? won't this leak this timer?

Reply:
Nothing ever calls it with disposing false, that only happens when you implement a custom deconstuctor.